src/checkin/services/flow/step/GetExtraPassengerInformationStep.ts

```diff
@@ -0,0 +1,42 @@
+import { Builder } from 'builder-pattern'
+import { Context } from '../../../model/Context'
+import { Session } from '../../../model/session'
+import { UserInformation } from '../../../model/session/UserInformation'
+import StepTemplate from '../StepTemplate'
+import { SessionData } from '../../../model/session/SessionData'
+
+export default class GetExtraPassengerInformationStep extends StepTemplate {
+
+  when(context: Context): boolean {
+    const numberOfPassengers = context.getSession().data.passengers
+    return numberOfPassengers > 1
+  }
+
+  onExecute(context: Context): Promise<boolean> {
+    const session = context.getSession()
+    const requestData = context.getRequest()
+    const amountOfExtraPassengers = session.data.passengers - 1
+    const extraPassengerInformation = requestData.fields?.extra_passenger_information as Array<{ name: string, passport: string }>
+
+    if (!extraPassengerInformation || amountOfExtraPassengers > extraPassengerInformation.length) {
+      context = context.withResponseBuilder((responseBuilder) => responseBuilder.status('user_information_required').requiredFiles({ extra_passenger_information: null }))
```

Comment:
The response with status { extra_passenger_information: null } could be improved in cases where the "extra_passenger_information" field is included in the request but not specified for all additional passengers (amountOfExtraPassengers > extraPassengerInformation.length). One option to provide more detailed feedback to the client could be returning something like:
```
{
  extra_passenger_information: {
    total_expected_extra_passengers: number,
    specified_extra_passenger_info_count: number
  }
}
```

Reply:
Sounds good, better than using null.

Thanks!